core/internal/features_test.go

```diff
@@ -951,17 +952,30 @@ func TestIntegration_FluxMonitor_Deviation(t *testing.T) {
 	mockServer, assertCalled := cltest.NewHTTPMockServer(t, http.StatusOK, "POST", priceResponse)
 	defer assertCalled()
 
+	confirmed := int64(23456)
+	safe := confirmed + int64(config.MinRequiredOutgoingConfirmations())
+	inLongestChain := safe - int64(config.GasUpdaterBlockDelay())
+
 	// Single task ethTx receives configuration from FM init and writes to chain.
-	attemptHash := cltest.NewHash()
+	gethClient.On("SubscribeFilterLogs", mock.Anything, mock.Anything, mock.Anything).
+		Return(logsSub, nil)
+	gethClient.On("FilterLogs", mock.Anything, mock.Anything).Return([]models.Log{}, nil)
 
 	// Initial tx attempt sent
-	rpcClient.On("Call", mock.Anything, "eth_sendRawTransaction", mock.Anything).
-		Run(func(args mock.Arguments) { *args.Get(0).(*common.Hash) = attemptHash }).
+	gethClient.On("SendTransaction", mock.Anything, mock.Anything).
+		Run(func(args mock.Arguments) {
+			tx, ok := args.Get(1).(*types.Transaction)
+			require.True(t, ok)
+			gethClient.On("TransactionReceipt", mock.Anything, mock.Anything).
+				Return(&types.Receipt{TxHash: tx.Hash(), BlockNumber: big.NewInt(confirmed)}, nil)
+		}).
+		Return(nil).Once()
+
+	rpcClient.On("CallContext", mock.Anything, mock.Anything, "eth_getBlockByNumber", mock.Anything, false).
+		Run(func(args mock.Arguments) { *args.Get(1).(**models.Head) = cltest.Head(inLongestChain) }). // TODO - RYAN - this is hacky
```

Comment:
Looks like it's in a couple places. It's a dense line. I'm not sure if there's a better way of doing it, but maybe it would be a little more comprehensible if broken up into a couple lines with descriptive variable names.